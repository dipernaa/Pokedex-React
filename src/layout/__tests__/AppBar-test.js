import React from 'react';
import { shallow } from 'enzyme';
import { browserHistory } from 'react-router';
import toJson from 'enzyme-to-json';
import mockStore from 'redux-mock-store';
import AppBar from '../AppBar';

jest.mock('react-router', () => ({
  browserHistory: {
    goBack: jest.fn()
  }
}));

const TEST_PROPS = {
  title: 'Super Title',
  showGlobalActions: jest.fn()
};

const mockReduxStore = mockStore({});

beforeEach(() => {
  browserHistory.goBack.mockClear();
  mockReduxStore.clearActions();
});

it('renders an AppBar', () => {
  const wrapper = shallow(
    <AppBar {...TEST_PROPS}/>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders an AppBar with a back arrow', () => {
  const props = {
    ...TEST_PROPS,
    showBackArrow: true
  };
  const wrapper = shallow(
    <AppBar {...props}/>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('calls browserHistory.goBack() when back arrow is pressed', () => {
  const props = {
    ...TEST_PROPS,
    showBackArrow: true
  };
  const wrapper = shallow(
    <AppBar {...props}/>
  );

  const appBarProps = wrapper.find('AppBar').props();
  appBarProps.onLeftIconButtonTouchTap();

  expect(browserHistory.goBack.mock.calls.length).toBe(1);
});
