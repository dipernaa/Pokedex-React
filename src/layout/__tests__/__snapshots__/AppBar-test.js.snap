// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an AppBar 1`] = `
<AppBar
  showMenuIconButton={false}
  title="Super Title"
  zDepth={1}
/>
`;

exports[`renders an AppBar with a back arrow 1`] = `
<AppBar
  iconElementLeft={
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <NavigationArrowBack />
    </IconButton>
  }
  onLeftIconButtonTouchTap={[Function]}
  showMenuIconButton={true}
  title="Super Title"
  zDepth={1}
/>
`;
